src/utils/coinFormatter.js

```diff
@@ -24,13 +24,13 @@ export const prettyBalance = (amount, coin, dp = VALUE_DECIMALS) => {
 }
 
 export const prettyFiatBalance = (amount, rate) => {
-  if (!amount || !rate) return amount
   const fiatAmount = cryptoToFiat(amount, rate)
+  if (isNaN(fiatAmount)) return fiatAmount
   return formatFiat(fiatAmount)
 }
 
 export const cryptoToFiat = (amount, rate) => {
-  if (!rate) return BN(amount)
+  if (!rate) return 'Rate missing'
```

Comment:
A tip from @kraikov. Replace this line with:
```
try {
   BigNumber(rate) 
} catch(err) { 
   return ''
}
```

Reply:
I think this is good, but it is not consistent with the other validations in  the file. 
Should we refactor all of them or just leave it as it is now with validation 

if (!rate) return 'Rate missing'